src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,510 @@
+        return svalue
+
+    def _set_heating(self, value):
+        """
+        Setter for the heater range VA
+        """
+        value = int(value)
+        self.SetHeaterRange(value)
+        # Read back the new value from the device
+        svalue = self.GetHeaterRange()
+        self.checkError()
+        if svalue != value:
+            logging.warning("Did not set new heating range to %d", value)
+        return svalue
+
+    def _poll(self):
+        '''
+        This method runs in a separate thread and polls the device for the temperature
+        '''
+        try:
+            temp = self.GetSensorTemperature()
+            self.temperature._set_value(temp, force_write=True)
+            logging.debug(u"Lakeshore temperature: %f °C", self.temperature.value)
+            self.checkError()
+        except Exception as e:
+            # another exception. End running.
+            logging.exception(e)
+
+
+class LakeshoreSimulator(object):
+    """
+    Simulates a Lakeshore 335
+    Same interface as the serial port
+    """
+
+    def __init__(self, timeout=1):
+        # we don't care about the actual parameters but timeout
+        self.timeout = timeout
+        self._output_buf = b""  # what the commands sends back to the "host computer"
+        self._input_buf = b""  # what we receive from the "host computer"
+        self._status_byte = POWER_ON
+
+        self._setpoint = 10
+        self._temperature = 50
+        self._heating = 1
+
+    def write(self, data):
+        self._input_buf += data
+        msgs = self._input_buf.split(b"\r")
+        for m in msgs[:-1]:
+            self._parseMessage(m)  # will update _output_buf
+
+        self._input_buf = msgs[-1]
+
+    def read(self, size=1):
+        ret = self._output_buf[:size]
+        self._output_buf = self._output_buf[len(ret):]
+
+        if len(ret) < size:
+            # simulate timeout
+            time.sleep(self.timeout)
+        return ret
+
+    def flush(self):
+        pass
+
+    def flushInput(self):
+        self._output_buf = b""
+
+    def close(self):
+        # using read or write will fail after that
+        del self._output_buf
+        del self._input_buf
+
+    def isOpen(self):
+        return hasattr(self, "_output_buf")
+
+    def _sendAnswer(self, ans):
+        self._output_buf += b"%s\r" % (ans,)
+
+    def _parseMessage(self, msg):
+        """
+        msg (str): the message to parse (without the \r)
+        return None: self._output_buf is updated if necessary
+        """
+        logging.debug("SIM: parsing %s", to_str_escape(msg))
+        msg = msg.strip()  # remove leading and trailing whitespace
+        msg = b"".join(msg.split())  # remove all space characters
+
+        if msg == b"*STB??":
+            self._sendAnswer(b"%d" % (self._status_byte,))
+        elif msg == b"*CLS":
+            self._status_byte = 0
+        elif msg == b"*IDN?":
+            self._sendAnswer(b"SIM,335,fake")
```

Comment:
Now the driver expects 4 values => Please make it look more like the real hardware.